### PR TITLE
Addition of 'locks' definitions.

### PR DESCRIPTION
**Pull request by flippynips**
- [x]  Use a meaningful title for the pull request. Include the name of the package modified.
- [x]   Test the change in your own code. (Compile and run.)
- [x]   Add tests. (Run with npm test.)
- [x]   Follow the advice from the readme.
- [x]   Avoid common mistakes.
- [x]   Run npm run lint package-name (or tsc if no tslint.json is present).\

I'm adding definitions of the npm package 'locks'; https://www.npmjs.com/package/locks

### CODE REVIEW

**Review by plantain-00 — APPROVED** [review]

File: types/locks/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+// Project: https://github.com/Wizcorp/locks
+// Definitions by: Joshua Graham <https://github.com/flippynips>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+/** Solitary access lock */
+export class Mutex {
+    /** Construct a new mutex lock. */
+    constructor();
+    /** Flag indicating whther the lock is currently taken. */
```
Comment (amcasey) [inline types/locks/index.d.ts]: Typo: "whether"

File: types/locks/locks-tests.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import * as locks from "locks";
```
Comment (amcasey) [inline types/locks/locks-tests.ts]: Since node-style exports are used, `import locks = require("locks");` might be preferable.

**Review by amcasey — APPROVED** [review]

File: types/locks/index.d.ts (hunk 1 above)
Comment (flippynips) [reply types/locks/index.d.ts]: done

File: types/locks/locks-tests.ts (hunk 2 above)
Comment (flippynips) [reply types/locks/locks-tests.ts]: and done!

**Pull request merged**